From foo@baz Tue Apr  9 12:12:43 2002
Date: Wed, 14 Jun 2006 12:14:34 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: USB: make usb_create_ep_files take a struct device

Instead of a kobject, will make things easier in the future (don't know
what I was thinking when I did this originally...)

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/endpoint.c |    4 ++--
 drivers/usb/core/sysfs.c    |    4 ++--
 drivers/usb/core/usb.h      |    2 +-
 3 files changed, 5 insertions(+), 5 deletions(-)

--- gregkh-2.6.orig/drivers/usb/core/endpoint.c
+++ gregkh-2.6/drivers/usb/core/endpoint.c
@@ -173,7 +173,7 @@ static struct kobj_type ep_object_ktype 
 	.default_attrs =	ep_attrs,
 };
 
-void usb_create_ep_files(struct kobject *parent,
+void usb_create_ep_files(struct device *parent,
 			 struct usb_host_endpoint *endpoint,
 			 struct usb_device *udev)
 {
@@ -189,7 +189,7 @@ void usb_create_ep_files(struct kobject 
 
 	kobj = &ep_obj->kobj;
 	kobject_set_name(kobj, "ep_%02x", endpoint->desc.bEndpointAddress);
-	kobj->parent = parent;
+	kobj->parent = &parent->kobj;
 	kobj->ktype = &ep_object_ktype;
 
 	/* Don't use kobject_register, because it generates a hotplug event */
--- gregkh-2.6.orig/drivers/usb/core/sysfs.c
+++ gregkh-2.6/drivers/usb/core/sysfs.c
@@ -223,7 +223,7 @@ void usb_create_sysfs_dev_files (struct 
 	if (udev->serial)
 		device_create_file (dev, &dev_attr_serial);
 	device_create_file (dev, &dev_attr_configuration);
-	usb_create_ep_files(&dev->kobj, &udev->ep0, udev);
+	usb_create_ep_files(dev, &udev->ep0, udev);
 }
 
 void usb_remove_sysfs_dev_files (struct usb_device *udev)
@@ -327,7 +327,7 @@ static inline void usb_create_intf_ep_fi
 
 	iface_desc = intf->cur_altsetting;
 	for (i = 0; i < iface_desc->desc.bNumEndpoints; ++i)
-		usb_create_ep_files(&intf->dev.kobj, &iface_desc->endpoint[i],
+		usb_create_ep_files(&intf->dev, &iface_desc->endpoint[i],
 				udev);
 }
 
--- gregkh-2.6.orig/drivers/usb/core/usb.h
+++ gregkh-2.6/drivers/usb/core/usb.h
@@ -4,7 +4,7 @@ extern void usb_create_sysfs_dev_files (
 extern void usb_remove_sysfs_dev_files (struct usb_device *dev);
 extern void usb_create_sysfs_intf_files (struct usb_interface *intf);
 extern void usb_remove_sysfs_intf_files (struct usb_interface *intf);
-extern void usb_create_ep_files(struct kobject *parent, struct usb_host_endpoint *endpoint,
+extern void usb_create_ep_files(struct device *parent, struct usb_host_endpoint *endpoint,
 				struct usb_device *udev);
 extern void usb_remove_ep_files(struct usb_host_endpoint *endpoint);
 
